quinn-proto/src/connection.rs

```diff
@@ -2172,7 +2172,7 @@ impl Connection {
             })
             .packet
             .tag_len();
-        let max_size = self.mtu as usize - tag_len;
+        let max_size = buf.capacity() - tag_len;
```

Comment:
I'm concerned that this makes it easy to inadvertently extend the buffer past `self.mtu`, if e.g. one of the guards in `populate_packet` has a bug. Maybe assert against that in `poll_transmit`?

Reply:
Done.

Reply:
On further consideration, that's a danger regardless of this change, so not a blocker.